api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -171,13 +172,11 @@ public void cleanBillingBuffer() {
         // filter as well as a first pass filter to limit what we pull into memory.
         .filter(entry -> entry.getLastStatusChangedTime() != null)
         .filter(entry -> entry.getLastSyncRequestTime() != null)
-        .filter(
-            entry ->
+        .filter(entry -> ComparableUtils.isGreaterThanOrEqualTo(
```

Comment:
I thought about switching this so it's a <= comparison, which are easier to read (especially if there are multiple ones).